docs/rules/no-restricted-syntax.md

```diff
@@ -0,0 +1,51 @@
+# Disallow certain syntax (no-restricted-syntax)
+
+Some code bases prefer to not use certain features of JavaScript such as `FunctionExpression` or `WithStatement`. This rule warns on the passed node types.
```

Comment:
Need more here. This needs to talk about ASTs and ping to the parser demo to help people figure out AST node types.